modin/core/storage_formats/pandas/utils.py

```diff
@@ -38,14 +38,14 @@ def get_default_chunksize(length, num_splits):
     )
 
 
-def compute_chunksize(df, num_splits, default_block_size=32, axis=None):
+def compute_chunksize(df_shape, num_splits, default_block_size=32, axis=None):
```

Comment:
Will split to another PR.

While you can see how this solution as a whole will look.